components/device-mgt/org.wso2.carbon.device.mgt.core/src/test/java/org/wso2/carbon/device/mgt/core/push/notification/mgt/task/PushNotificationSchedulerTaskTest.java

```diff
@@ -0,0 +1,89 @@
+import org.wso2.carbon.device.mgt.core.service.DeviceManagementProviderServiceImpl;
+import org.wso2.carbon.registry.core.exceptions.RegistryException;
+
+import java.sql.SQLException;
+import java.util.List;
+import java.util.Map;
+
+public class PushNotificationSchedulerTaskTest extends BaseDeviceManagementTest {
+    private static final Log log = LogFactory.getLog(PushNotificationSchedulerTask.class);
+    private DeviceManagementProviderService deviceMgtProviderService;
+    private PushNotificationSchedulerTask pushNotificationSchedulerTask;
+    private OperationDAO operationDAO;
+
+    @BeforeClass
+    public void init() throws DeviceManagementException, RegistryException {
+        DeviceConfigurationManager.getInstance().initConfig();
+        log.info("Initializing Push Notification Scheduler Test Class");
+        DeviceManagementServiceComponent.notifyStartupListeners();
+        this.deviceMgtProviderService = Mockito.mock(DeviceManagementProviderServiceImpl.class, Mockito.CALLS_REAL_METHODS);
+        DeviceManagementDataHolder.getInstance().setDeviceManagementProvider(this.deviceMgtProviderService);
+        this.operationDAO = OperationManagementDAOFactory.getOperationDAO();
+        this.pushNotificationSchedulerTask = new PushNotificationSchedulerTask();
+    }
+
+    @Test
+    public void testPushNotificationScheduler() {
+        try {
+            log.debug("Attempting to execute push notification task scheduler");
+            Mockito.doReturn(new TestNotificationStrategy()).when(this.deviceMgtProviderService)
+                    .getNotificationStrategyByDeviceType(Mockito.anyString());
+            Mockito.doReturn(new org.wso2.carbon.device.mgt.common.operation.mgt.Operation())
+                    .when(this.deviceMgtProviderService).getOperation(Mockito.anyString(), Mockito.anyInt());
+            this.pushNotificationSchedulerTask.run();
+            OperationManagementDAOFactory.openConnection();
+            Map<Integer, List<OperationMapping>> operationMappingsTenantMap = operationDAO
+                    .getOperationMappingsByStatus(Operation.Status.PENDING, Operation.PushNotificationStatus.SCHEDULED,
+                            DeviceConfigurationManager.getInstance().getDeviceManagementConfig()
+                                    .getPushNotificationConfiguration().getSchedulerBatchSize());
+            Assert.assertEquals(operationMappingsTenantMap.size(), 0);
+            log.debug("Push notification task execution complete.");
+        } catch (DeviceManagementException e) {
+            Assert.fail("Unexpected exception occurred when getting the push notification strategy.", e);
+        } catch (SQLException | OperationManagementDAOException e) {
+            Assert.fail("Unexpected exception occurred retrieving the operation mapping list.", e);
+        } catch (OperationManagementException e) {
+            Assert.fail("Unexpected exception occurred when retrieving an operation.", e);
```

Comment:
We do not need to fail the test cases like this. We can simply throw the exceptions by adding them to method signature. If any of those exceptions are thrown, there will be test case failure automatically